README.md

```diff
@@ -151,18 +156,37 @@ The repository is organized in six Gradle projects:
  * `realm-transformer`: it contains the bytecode transformer.
  * `gradle-plugin`: it contains the Gradle plugin.
  * `examples`: it contains the example projects. This project directly depends on `gradle-plugin` which adds a dependency to the artifacts produced by `realm`.
- * The root folder is another Gradle project and all it does is orchestrating the other jobs
+ * The root folder is another Gradle project.  All it does is orchestrate the other jobs
 
 This means that `./gradlew clean` and `./gradlew cleanExamples` will fail if `assembleExamples` has not been executed first.
 Note that IntelliJ [does not support multiple projects in the same window](https://youtrack.jetbrains.com/issue/IDEABKL-6118#)
-so each sub-project must be opened in its own window.
+so each of the six Gradle projects must be imported as a separate IntelliJ project.
+
+Since the repository contains several completely independent Gradle projects, several independent builds are run to assemble it.
+Seeing a line like: `:realm:realm-library:compileBaseDebugAndroidTestSources UP-TO-DATE` in the build log does *not* imply
+that you can run `./gradlew :realm:realm-library:compileBaseDebugAndroidTestSources`.
 
 ## Examples
 
 The `./examples` folder contain a number of example projects showing how Realm can be used. If this is the first time you checkout or pull a new version of this repository to try the examples, you must call `./gradlew installRealmJava` from the top-level directory first. Otherwise the examples will not compile as they depend on all Realm artifacts being installed in `mavenLocal()`.
 
 Standalone examples can be [downloaded from website](https://realm.io/docs/java/latest/#getting-started).
 
+## Running Tests on a Device
+
+To run these tests you must have a device connected the the build computer and the `adb` command must be in your `PATH`
+
+1. Connect an Android device and verify that that the command `adb shell` starts a shell session on the device.
```

Comment:
why do we need to start a shell session?
a successful `adb devices` where the user can see his device will be enough IMO